tools/build-toolchain.sh

```diff
@@ -34,139 +31,273 @@ export PATH="$PATH:$INSTALL_PATH/bin"
 JOBS="${JOBS:-$(getconf _NPROCESSORS_ONLN)}"
 JOBS="${JOBS:-1}" # If getconf returned nothing, default to 1
 
-# Additional GCC configure arguments
+# GCC configure arguments to use system GMP/MPC/MFPF
 GCC_CONFIGURE_ARGS=()
 
 # Dependency source libs (Versions)
 BINUTILS_V=2.39
 GCC_V=12.2.0
 NEWLIB_V=4.2.0.20211231
+GMP_V=6.2.0 
+MPC_V=1.2.1 
+MPFR_V=4.1.0
+MAKE_V=${MAKE_V:-""}
 
 # Check if a command-line tool is available: status 0 means "yes"; status 1 means "no"
 command_exists () {
-  (command -v "$1" >/dev/null 2>&1)
-  return $?
+    (command -v "$1" >/dev/null 2>&1)
+    return $?
 }
 
 # Download the file URL using wget or curl (depending on which is installed)
 download () {
-  if   command_exists wget ; then wget -c  "$1"
-  elif command_exists curl ; then curl -LO "$1"
-  else
-    echo "Install wget or curl to download toolchain sources" 1>&2
-    return 1
-  fi
+    if   command_exists wget ; then wget -c  "$1"
+    elif command_exists curl ; then curl -LO "$1"
+    else
+        echo "Install wget or curl to download toolchain sources" 1>&2
+        return 1
+    fi
 }
 
 # Compilation on macOS via homebrew
 if [[ $OSTYPE == 'darwin'* ]]; then
-  if ! command_exists brew; then
-    echo "Compilation on macOS is supported via Homebrew (https://brew.sh)"
-    echo "Please install homebrew and try again"
-    exit 1
-  fi
+    if ! command_exists brew; then
+        echo "Compilation on macOS is supported via Homebrew (https://brew.sh)"
+        echo "Please install homebrew and try again"
+        exit 1
+    fi
 
-  # Install required dependencies
-  brew install -q gmp mpfr libmpc gsed
+    # Install required dependencies. gsed is really required, the others are optionals
+    # and just speed up build.
+    brew install -q gmp mpfr libmpc gsed
 
-  # Tell GCC configure where to find the dependent libraries
-  GCC_CONFIGURE_ARGS=(
-    "--with-gmp=$(brew --prefix)"
-    "--with-mpfr=$(brew --prefix)"
-    "--with-mpc=$(brew --prefix)"
-  )
+    # Tell GCC configure where to find the dependent libraries
+    GCC_CONFIGURE_ARGS=(
+        "--with-gmp=$(brew --prefix)"
+        "--with-mpfr=$(brew --prefix)"
+        "--with-mpc=$(brew --prefix)"
+    )
 
-  # Install GNU sed as default sed in PATH. GCC compilation fails otherwise,
-  # because it does not work with BSD sed.
-  export PATH="$(brew --prefix gsed)/libexec/gnubin:$PATH"
+    # Install GNU sed as default sed in PATH. GCC compilation fails otherwise,
+    # because it does not work with BSD sed.
+    PATH="$(brew --prefix gsed)/libexec/gnubin:$PATH"
+    export PATH
 fi
 
 # Create build path and enter it
 mkdir -p "$BUILD_PATH"
 cd "$BUILD_PATH"
 
-# Dependency source: Download stage
+# Dependency downloads and unpack
 test -f "binutils-$BINUTILS_V.tar.gz" || download "https://ftp.gnu.org/gnu/binutils/binutils-$BINUTILS_V.tar.gz"
+test -d "binutils-$BINUTILS_V"        || tar -xzf "binutils-$BINUTILS_V.tar.gz"
```

Comment:
Perhaps use `-C` to unpack to expected directory... (at least for debug purposes...)